core/templates/services/math-interactions.service.ts

```diff
@@ -96,6 +96,46 @@ export class MathInteractionsService {
     return errorMessage;
   }
 
+  isParenRedundant(
+      expressionString: string,
+      openingInd: number,
+      closingInd: number): boolean {
+    /*
+    Assumes that expressionString is syntactically valid.
+
+    Multiple consecutive parens are considered redundant. eg: for ((a - b))
+    the outer pair of parens are considered as redundant.
+    */
+    let leftParenIsRedundant = (
+      openingInd - 1 >= 0 && expressionString[openingInd - 1] === '(');
+    let rightParenIsRedundant = (
+      closingInd + 1 < expressionString.length &&
+      expressionString[closingInd + 1] === ')');
+    return leftParenIsRedundant && rightParenIsRedundant;
+  }
+
+  containsRedundantParens(expressionString: string): [boolean, string] {
```

Comment:
Could you please add a docstring explaining what the components of the return value represent?

Reply:
Added.